lib/mutant/repository/diff.rb

```diff
@@ -31,15 +31,21 @@ def touches?(path, line_range)
 
       # rubocop:disable Metrics/MethodLength
       def touched_paths
-        pathname = world.pathname
-        work_dir = pathname.pwd
+        root_stdout =
+          world
+            .capture_stdout(%w[git rev-parse --show-toplevel])
+            .from_right
+            .lines
+            .first
+            .strip
```

Comment:
Note that `String#strip` is dangerous in this context:


```
irb(main):003:0> " /some/path\n".strip
=> "/some/path"
irb(main):004:0> " /some/path\n".chomp
=> " /some/path"
```

It could be argued that the extra whitespace in the beggining of the string should "never" exist. Hence its good to filter it out. But to me "silently" filtering it out is even more an issue than letting it bubble up, where it can produce an explicit error on followup operations.

My version in #1142 uses `String#chomp` for that reason. Which arguably is still "bad" as `String#chomp` silently does accept this `"/some/path"` (without the expected newline).

The real solution is to parse the path, and enforce the presence of the expected newline and fail in all other cases.

For now, given ruby is such a "high overhead" language I'll go with `String#chomp`.

Reply:
👍